crates/aptos-protos/proto/aptos/indexer/v1/indexer_data.proto

```diff
@@ -22,20 +22,42 @@ message GetTransactionsRequest {
 
   // Optional; number of transactions to return in current batch.
   optional uint64 batch_size = 3;
+
+  // Opinoal; status of the processing is required in the response. 
+  optional bool is_status_required = 4 [default = false];
+}
+
+// DataStatus is an optional response from the server to the client.
+// It can benefit the downstream:
+//   1. The client can get a heartbeat-like reponse from the server.
+//   2. The client can track the last version processed by the server with a timely manner.
+// This is a common signal frame that can be used by all the streaming RPCs.
+message DataStatus {
```

Comment:
I wonder if this is the best way to send health check data? Is the idea that if `is_status_required` flag is true, then we will send a `DataStatus` message every X seconds. 

If the client specifies a filter, will the `DataStatus` contain the last version processed _without_ the filter? I think that could work. Then the client will update that value in their last_processed_version cursor.

BTW there are other ways to build health-checking protocols, an example I found: https://github.com/grpc/grpc/blob/master/doc/health-checking.md

Reply:
good call. let's take this discussion to the `custom parsing` design and you can drive this haha